### PR TITLE
*: switch from Fixed8 to int64

### PR DESCRIPTION
**Pull request by AnnaShaleva**

### CODE REVIEW

**Comment by fyrchik** [comment]
I am now thinking about the purpose of `decimals` of `NEP5`.

File: pkg/rpc/server/server_helper_test.go [hunk 1]
```diff
@@ -85,10 +85,10 @@ func initServerWithInMemoryChain(t *testing.T) (*core.Blockchain, *Server, *http
...
 
-func (fs *FeerStub) FeePerByte() util.Fixed8 {
+func (fs *FeerStub) FeePerByte() int64 {
 	return 0
 }
 
-func (fs *FeerStub) GetUtilityTokenBalance(acc util.Uint160) util.Fixed8 {
-	return util.Fixed8FromInt64(1000000)
+func (fs *FeerStub) GetUtilityTokenBalance(acc util.Uint160) int64 {
+	return 100000000000000
```
Comment (fyrchik) [inline pkg/rpc/server/server_helper_test.go:93]: It is worth to have 'native.GASFactor` or smth similar.

File: cli/wallet/nep5.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ func newNEP5Commands() []cli.Command {
 			Name:  "amount",
 			Usage: "Amount of asset to send",
 		},
-		flags.Fixed8Flag{
+		cli.Int64Flag{
```
Comment (roman-khimov) [inline cli/wallet/nep5.go:59]: That's not user-friendly. I think we need to keep the old notation here. For user-facing things there is a `BigDecimal` in C# node.

File: pkg/core/blockchain.go [hunk 3]
```diff
@@ -788,23 +788,23 @@ func (bc *Blockchain) GetNEP5Balances(acc util.Uint160) *state.NEP5Balances {
 }
 
 // GetUtilityTokenBalance returns utility token (GAS) balance for the acc.
-func (bc *Blockchain) GetUtilityTokenBalance(acc util.Uint160) util.Fixed8 {
+func (bc *Blockchain) GetUtilityTokenBalance(acc util.Uint160) int64 {
```
Comment (roman-khimov) [inline pkg/core/blockchain.go:791]: OK for now, but #1133.

File: pkg/core/helper_test.go [hunk 4]
```diff
@@ -210,7 +210,7 @@ func TestCreateBasicChain(t *testing.T) {
 	t.Logf("txMoveNeo: %s", txMoveNeo.Hash().StringLE())
 	t.Logf("txMoveGas: %s", txMoveGas.Hash().StringLE())
 
-	require.True(t, util.Fixed8FromInt64(1000).CompareTo(bc.GetUtilityTokenBalance(priv0ScriptHash)) <= 0)
+	require.True(t, bc.GetUtilityTokenBalance(priv0ScriptHash) >= 100000000000)
```
Comment (roman-khimov) [inline pkg/core/helper_test.go:213]: It's hard to count the number of zeroes here.

File: pkg/core/helper_test.go [hunk 5]
```diff
@@ -241,8 +241,7 @@ func TestCreateBasicChain(t *testing.T) {
 	emit.Syscall(script.BinWriter, "System.Contract.Create")
 	txScript := script.Bytes()
 
-	invFee := util.Fixed8FromFloat(100)
-	txDeploy := transaction.New(testchain.Network(), txScript, invFee)
+	txDeploy := transaction.New(testchain.Network(), txScript, 10000000000)
```
Comment (roman-khimov) [inline pkg/core/helper_test.go:244]: And here.

File: pkg/core/interop/crypto/ecdsa.go [hunk 6]
```diff
@@ -39,7 +38,7 @@ func ECDSACheckMultisig(ic *interop.Context, v *vm.VM) error {
 	if err != nil {
 		return fmt.Errorf("wrong parameters: %s", err.Error())
 	}
-	if !v.AddGas(util.Fixed8(ECDSAVerifyPrice * len(pkeys))) {
+	if !v.AddGas(int64(ECDSAVerifyPrice * len(pkeys))) {
```
Comment (roman-khimov) [inline pkg/core/interop/crypto/ecdsa.go:41]: I'd better do `ECDSAVerifyPrice * int64(len(pkeys))`.

File: pkg/core/util.go [hunk 7]
```diff
@@ -29,7 +29,7 @@ var (
 
 	// ecdsaVerifyInteropPrice returns the price of Neo.Crypto.ECDsaVerify
 	// syscall to calculate NetworkFee for transaction
-	ecdsaVerifyInteropPrice = util.Fixed8(100000)
+	ecdsaVerifyInteropPrice int64 = 100000
```
Comment (roman-khimov) [inline pkg/core/util.go:32]: Should be a `const` now.

File: pkg/rpc/rpc_config.go [hunk 8]
```diff
@@ -10,9 +8,9 @@ type (
 		EnableCORSWorkaround bool   `yaml:"EnableCORSWorkaround"`
 		// MaxGasInvoke is a maximum amount of gas which
 		// can be spent during RPC call.
-		MaxGasInvoke util.Fixed8 `yaml:"MaxGasInvoke"`
-		Port         uint16      `yaml:"Port"`
-		TLSConfig    TLSConfig   `yaml:"TLSConfig"`
+		MaxGasInvoke int64     `yaml:"MaxGasInvoke"`
```
Comment (roman-khimov) [inline pkg/rpc/rpc_config.go:11]: I'm a user and I want to specify `10` GAS in my config. Not `1000000000` of something.

File: pkg/core/util.go [hunk 9]
```diff
@@ -156,6 +156,6 @@ func calculateMultisigFee(n int) int64 {
 	bw := io.NewBufBinWriter()
 	emit.Int(bw.BinWriter, int64(n))
 	// it's a hack because prices of small PUSH* opcodes are equal
-	result = result + opcodePrice(opcode.Opcode(bw.Bytes()[0]))
+	result += opcodePrice(opcode.Opcode(bw.Bytes()[0]))
```
Comment (roman-khimov) [inline pkg/core/util.go:159]: Not a part of 3d04fd106d02ec8701f2740c6add15961cb2ead9.

File: pkg/core/util.go [hunk 10]
```diff
@@ -138,13 +138,13 @@ func CalculateNetworkFee(script []byte) (int64, int) {
...
-		netFee = netFee + opcodePrice(opcode.PUSHDATA1, opcode.PUSHNULL) + ecdsaVerifyInteropPrice
+		netFee += opcodePrice(opcode.PUSHDATA1, opcode.PUSHNULL) + ecdsaVerifyInteropPrice
 	} else if n, pubs, ok := vm.ParseMultiSigContract(script); ok {
 		m := len(pubs)
 		sizeInv := 66 * m
 		size += io.GetVarSize(sizeInv) + sizeInv + io.GetVarSize(script)
-		netFee = netFee + calculateMultisigFee(m) + calculateMultisigFee(n)
-		netFee = netFee + opcodePrice(opcode.PUSHNULL) + int64(ecdsaVerifyInteropPrice)*int64(n)
+		netFee += calculateMultisigFee(m) + calculateMultisigFee(n)
+		netFee += opcodePrice(opcode.PUSHNULL) + ecdsaVerifyInteropPrice*int64(n)
```
Comment (roman-khimov) [inline pkg/core/util.go:147]: Not a part of 3d04fd106d02ec8701f2740c6add15961cb2ead9.

File: pkg/core/util.go [hunk 11]
```diff
@@ -138,13 +138,13 @@ func CalculateNetworkFee(script []byte) (int64, int) {
 	)
 	if vm.IsSignatureContract(script) {
 		size += 67 + io.GetVarSize(script)
-		netFee = netFee + opcodePrice(opcode.PUSHDATA1, opcode.PUSHNULL) + ecdsaVerifyInteropPrice
+		netFee += opcodePrice(opcode.PUSHDATA1, opcode.PUSHNULL) + ecdsaVerifyInteropPrice
```
Comment (roman-khimov) [inline pkg/core/util.go:141]: Not a part of 3d04fd106d02ec8701f2740c6add15961cb2ead9.

File: pkg/consensus/consensus_test.go [hunk 12]
```diff
@@ -263,9 +263,9 @@ func signTx(t *testing.T, feePerByte int64, txs ...*transaction.Transaction) {
 	for _, tx := range txs {
 		size := io.GetVarSize(tx)
 		netFee, sizeDelta := core.CalculateNetworkFee(rawScript)
-		tx.NetworkFee = tx.NetworkFee + netFee
+		tx.NetworkFee += netFee
 		size += sizeDelta
-		tx.NetworkFee = tx.NetworkFee + (int64(size) * feePerByte)
+		tx.NetworkFee += int64(size) * feePerByte
```
Comment (roman-khimov) [inline pkg/consensus/consensus_test.go:268]: Not a part of 3d04fd106d02ec8701f2740c6add15961cb2ead9.

File: pkg/core/blockchain.go [hunk 13]
```diff
@@ -1062,8 +1062,8 @@ func (bc *Blockchain) UnsubscribeFromExecutions(ch chan<- *state.AppExecResult)
 // amount of NEO between specified blocks. The amount of NEO being passed is in
 // its natural non-divisible form (1 NEO as 1, 2 NEO as 2, no multiplication by
 // 10⁸ is neeeded as for Fixed8).
-func (bc *Blockchain) CalculateClaimable(value int64, startHeight, endHeight uint32) util.Fixed8 {
-	var amount util.Fixed8
+func (bc *Blockchain) CalculateClaimable(value int64, startHeight, endHeight uint32) int64 {
+	var amount int64
```
Comment (roman-khimov) [inline pkg/core/blockchain.go:1066]: Some `CalculateClaimable` user are probably need to be updated too now, but I don't see that in 3d04fd106d02ec8701f2740c6add15961cb2ead9.

File: pkg/rpc/server/server_helper_test.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/rpc/server/server_helper_test.go]: Fixed by using `native.GASFactor`.

File: pkg/core/helper_test.go (hunk 4 above)
Comment (AnnaShaleva) [reply pkg/core/helper_test.go]: Fixed by using `native.GASFactor`.

File: pkg/core/util.go (hunk 9 above)
Comment (AnnaShaleva) [reply pkg/core/util.go:159]: That was unsuccessful rebase. Fixed.

File: pkg/core/util.go [hunk 14]
```diff
@@ -17,6 +17,10 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
 )
 
+// ecdsaVerifyInteropPrice returns the price of Neo.Crypto.ECDsaVerify
+// syscall to calculate NetworkFee for transaction
+const ecdsaVerifyInteropPrice int64 = 100000
```
Comment (roman-khimov) [inline pkg/core/util.go:22]: I think even `int64` could be dropped here, it's not really needed.

File: pkg/rpc/server/server.go [hunk 15]
```diff
@@ -786,8 +786,8 @@ func (s *Server) getUnclaimedGas(ps request.Params) (interface{}, *response.Erro
 	if neo == 0 {
 		return "0", nil
 	}
-	gas := s.chain.CalculateClaimable(int64(neo), neoHeight, s.chain.BlockHeight()+1) // +1 as in C#, for the next block.
-	return strconv.FormatInt(int64(gas), 10), nil                                     // It's not represented as Fixed8 in C#.
+	gas := s.chain.CalculateClaimable(neo, neoHeight, s.chain.BlockHeight()+1) // +1 as in C#, for the next block.
+	return strconv.FormatInt(int64(gas), 10), nil                              // It's not represented as Fixed8 in C#.
```
Comment (roman-khimov) [inline pkg/rpc/server/server.go]: And this comment is no longer relevant as it's not Fixed8 in neo-go either.

**Pull request merged**